crates/pallet-domains/src/runtime_registry.rs

```diff
@@ -134,19 +94,26 @@ pub(crate) fn register_runtime_at_genesis<T: Config>(
 /// Schedules a runtime upgrade after `DomainRuntimeUpgradeDelay` from current block number.
 pub(crate) fn do_schedule_runtime_upgrade<T: Config>(
     runtime_id: RuntimeId,
-    code: Vec<u8>,
+    raw_genesis_storage: Vec<u8>,
```

Comment:
we already have the raw_genesis when we register a runtime. Why do we need this again ?

Reply:
Because the genesis state will be changed as runtime version changes as observed previously, so every different runtime needs a corresponding genesis storage.

Reply:
This doesn't seem right. If we upgrade existing domain, we shouldn't wipe the state and replace it with one from genesis. Domain should do the actual upgrade instead with provided updated code.

Reply:
> Because the genesis state will be changed as runtime version changes as observed previously, so every different runtime needs a corresponding genesis storage.

^

@nazar-pc 's comment is wrong, plz take a look at https://github.com/subspace/subspace/pull/1824#discussion_r1326576232

Reply:
Passing genesis storage means you can instantitate domain with correct storage right away, otherwise you'll need to do multiple runtime upgrades right after instantiation, which makes little sense.